src/storage/types.rs

```diff
@@ -225,7 +225,7 @@ impl Clone for Key {
 
 impl Display for Key {
     fn fmt(&self, f: &mut Formatter) -> fmt::Result {
-        write!(f, "{}", hex::encode_upper(&self.0))
+        self.0.write_hex_upper(f)
```

Comment:
How does this change eliminate a key copy?

Reply:
Eliminating a string buffer in `hex::encode_upper()`:
```
pub fn encode_upper<T: AsRef<[u8]>>(data: T) -> String {
    let mut s = String::with_capacity(data.as_ref().len() * 2);

    // Writing to a string never errors, so we can unwrap here.
    data.write_hex_upper(&mut s).unwrap();
    s
}
```

Reply:
Nice finding!